src/Security/Authentication/test/TwitterTests.cs

```diff
@@ -273,6 +273,81 @@ public async Task BadCallbackCallsAccessDeniedWithState()
             Assert.Equal(HttpStatusCode.NotAcceptable, response.StatusCode);
         }
 
+        [Fact]
+        public async Task TwitterError_Json_ThrowsParsedException()
+        {
+            using var host = await CreateHost(o =>
+            {
+                o.ConsumerKey = "Test Consumer Key";
+                o.ConsumerSecret = "Test Consumer Secret";
+                o.BackchannelHttpHandler = new TestHttpMessageHandler
+                {
+                    Sender = JsonErroredBackchannelRequestToken
+                };
+            },
+            async context =>
+            {
+                await context.ChallengeAsync("Twitter");
+                return true;
+            });
+            using var server = host.GetTestServer();
+            
+            await Assert.ThrowsAsync<InvalidOperationException>(async () =>
```

Comment:
This returns the caught exception. Add an assert for the expected exception message.